hscontrol/oidc.go

```diff
@@ -157,15 +164,32 @@ func (a *AuthProviderOIDC) RegisterHandler(
 
 	stateStr := hex.EncodeToString(randomBlob)[:32]
 
-	// place the node key into the state cache, so it can be retrieved later
-	a.registrationCache.Set(
-		stateStr,
-		machineKey,
-	)
+	// Initialize registration info with machine key
+	registrationInfo := RegistrationInfo{
+		MachineKey: machineKey,
+	}
 
-	// Add any extra parameter provided in the configuration to the Authorize Endpoint request
-	extras := make([]oauth2.AuthCodeOption, 0, len(a.cfg.ExtraParams))
+	// Pre-allocate extras slice with known capacity
+	extrasCapacity := len(a.cfg.ExtraParams)
+	if a.cfg.EnablePKCE {
+		extrasCapacity += 2 // For AccessTypeOffline and S256ChallengeOption
+	}
+	extras := make([]oauth2.AuthCodeOption, 0, extrasCapacity)
+
+	// Add PKCE verification if enabled
+	if a.cfg.EnablePKCE {
+		verifier := oauth2.GenerateVerifier()
+		registrationInfo.Verifier = &verifier
+		extras = append(extras,
+			oauth2.AccessTypeOffline,
+			oauth2.S256ChallengeOption(verifier),
+		)
+	}
```

Comment:
This prealloc logic seems a bit unnecessary from a readability vs actual performance impact, I'm in favour of a simpler implementation:

```suggestion
	var extras []oauth2.AuthCodeOption

	// Add PKCE verification if enabled
	if a.cfg.EnablePKCE {
		verifier := oauth2.GenerateVerifier()
		registrationInfo.Verifier = &verifier
		extras = append(extras,
			oauth2.AccessTypeOffline,
			oauth2.S256ChallengeOption(verifier),
		)
	}
```